packages/core/test/emulate/mockReportingObserver.ts

```diff
@@ -50,8 +50,16 @@ export function mockReportingObserver() {
 export type MockCspEventListener = ReturnType<typeof mockCspEventListener>
 
 export function mockCspEventListener() {
-  spyOn(document, 'addEventListener').and.callFake((_type: string, listener: EventListener) => {
-    listeners.push(listener)
+  // eslint-disable-next-line @typescript-eslint/unbound-method
+  const originalAddEventListener = EventTarget.prototype.addEventListener
+  EventTarget.prototype.addEventListener = jasmine
+    .createSpy()
```

Comment:
This seems unrelated to this PR, but in any case:

**❓ question:** ‏`spyOn(EventTarget.prototype, 'addEventListener')` didn't work?

Reply:
I found cases where mocking the same API both manually with `registerCleanup` and with `spyOn` case don't work. I can't really explain why, but it seems reasonable to use the same mocking strategy twice. wdyt?

Reply:
> wdyt?

I guess spies are removed after all `afterEach`, so we have an order issue similar to what you are fixing in this ticket. So, I guess it makes sense.

Your explanation makes sense but it's a bit annoying that we cannot use Jasmine tools for that. This sounds extremely error prone..

One workaround I have in mind is to always attach spies first, before any other instrumentation. Then they will be detached after everything, so it will be fine.

An alternative could be to have our own `spyOn` utility that works with `registerCleanupTask` and forbid the jasmine one.